py3status/modules/alpine_updates.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+"""
+Display number of updates and more for Alpine Linux.
+
+Configuration parameters:
+    cache_timeout: refresh interval for this module (default 3600)
+    format: display format for this module (default 'UPD {count}')
+    format_separator: show separator if more than one (default None)
+    format_update: display format for updates (default None)
+    thresholds: specify color thresholds to use (default [])
+
+Format placeholders:
+    {count} number of updates
+    {format_update} format for updates
+
+format_update placeholders:
+    {name} package name, eg py3status
+    {new_version} package version, eg 3.7.0
+    {new_release} package release, eg 1
+    {old_version} package old version, 3.6.0
+    {old_release} package old release, eg 0
+
+Color thresholds:
+    count: a color based on number of updates
+
+Requires:
+    apk needs root privileges to update repository indexes from remote
+    repositories. We can schedule a hourly cron using the file below.
+
+    | /etc/periodic/hourly/py3status-apk_updates
+    | -------------------------------------------
+    | #!/bin/sh
+    | /sbin/apk update --quiet
+
+@author lasers
+
+Examples:
+```
+# add colors
+alpine_updates {
+    color_alpine = '#0D597F'
+}
+
+# show UPDATE
+alpine_updates {
+    format = '\?if=count&color=alpine UPDATE'
+}
+
+# show update shield with a count
+alpine_updates {
+    format = '[\?not_zero [\?color=alpine&show \u26ca] {count}]'
+}
+
+# show count and names
+alpine_updates {
+    format = '[\?not_zero&color=alpine UPD {count}][: {format_update}]'
+    format_update = '\?color=alpine {name}'
+}
+
+# show update bars
+alpine_updates {
+    format = '{format_update}'
+    format_update = '[\?color=alpine \|]'
+}
+
+# show count thresholds
+alpine_updates {
+    format = '[\?not_zero [\?color=count&show UPD] {count}]'
+    thresholds = [(0, 'good'), (20, 'degraded'), (30, 'bad')]
+}
+```
+
+SAMPLE OUTPUT
+{'full_text': 'UPD 13'}
+"""
+
+STRING_NOT_INSTALLED = 'apk not installed'
+APK_COMMAND = ['apk', 'version', '-l', '"<"']
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    cache_timeout = 3600
+    format = 'UPD {count}'
+    format_separator = None
+    format_update = None
+    thresholds = []
+
+    def post_config_hook(self):
+        if not self.py3.check_commands(APK_COMMAND[0]):
+            raise Exception(STRING_NOT_INSTALLED)
+        if not self.format_separator:
+            self.format_separator = ''
+
+    def alpine_updates(self):
+        apk_data = self.py3.command_output(APK_COMMAND).splitlines()[1:]
+        count = len(apk_data)
+        format_update = None
+
+        if self.format_update and apk_data:
+            new_data = []
+            for line in apk_data:
+                package = [x.strip() for x in line.split('=')]
```

Comment:
Ok, I had a chance to test it. This part didn't work, because this is what one package string looks like when you do
`apk version -l "<"`:

```
['wayland-1.13.0-r0                       < 1.14.0-r2']
```

I was able to fix it like this:

```
package_old, package_new = [x.strip() for x in line.split('<')]
new_version, new_release = package_new.rsplit('-r')
pkg_string, old_release = package_old.rsplit('-r')
name, old_version = pkg_string.rsplit('-', 1)
```

And it works now 🎉 

![2017-11-11-031206_1024x768_scrot](https://user-images.githubusercontent.com/17767357/32687708-78879250-c690-11e7-9742-fae55f58e246.png)

Reply:
@igorgrebenkov Thanks for testing this. New commit coming up.